cpp/src/skyhook/CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitationsn
+# under the License.
+
+#
+# arrow_skyhook
+#
+# define project properties
+project(arrow_skyhook)
+
+# add the client subdirectory
+add_subdirectory(client)
+
+# define the targets to build
+add_custom_target(arrow_skyhook_client)
+add_custom_target(cls_skyhook)
+
+# define the dependencies
+find_package(librados REQUIRED)
+include_directories(${LIBRADOS_INCLUDE_DIR})
+set(ARROW_SKYHOOK_LINK_STATIC arrow_dataset_static)
+set(ARROW_SKYHOOK_LINK_SHARED arrow_dataset_shared)
+set(ARROW_SKYHOOK_LINK_STATIC ${ARROW_SKYHOOK_LINK_STATIC} ${LIBRADOS_LIBRARY})
+set(ARROW_SKYHOOK_LINK_SHARED ${ARROW_SKYHOOK_LINK_SHARED} ${LIBRADOS_LIBRARY})
+
+# define the client and cls sources
+set(ARROW_SKYHOOK_CLIENT_SOURCES client/file_skyhook.cc protocol/rados_protocol.cc
+                                 protocol/skyhook_protocol.cc)
+set(ARROW_SKYHOOK_CLS_SOURCES cls/cls_skyhook.cc protocol/rados_protocol.cc
+                              protocol/skyhook_protocol.cc)
+
+# define the client library
+add_arrow_lib(arrow_skyhook_client
+              SOURCES
+              ${ARROW_SKYHOOK_CLIENT_SOURCES}
+              OUTPUTS
+              ARROW_SKYHOOK_CLIENT_LIBRARIES
+              SHARED_LINK_LIBS
+              ${ARROW_SKYHOOK_LINK_SHARED}
+              STATIC_LINK_LIBS
+              ${ARROW_SKYHOOK_LINK_STATIC})
```

Comment:
Could you add `PKG_CONFIG_NAME skyhook` and add `cpp/src/skyhook/skyhook.pc.in`?

See also:

* https://github.com/apache/arrow/blob/master/cpp/src/gandiva/CMakeLists.txt#L125-L126
* https://github.com/apache/arrow/blob/master/cpp/src/gandiva/gandiva.pc.in